no-vary-query.md

```diff
@@ -0,0 +1,264 @@
+
+To resolve this, we specify that prerendering [activation](https://wicg.github.io/nav-speculation/prerendering.html#prerendering-browsing-context-activate) involves changing the page's URL before firing the `prerenderingchange` event. This URL change is done using the [URL and history update steps](https://html.spec.whatwg.org/#url-and-history-update-steps), i.e. the same mechanism underlying `history.replaceState()`.
+
+With this setup, prerendered pages which care about a given query parameter, but don't want mismatches to prevent a prerender, can delay processing of the query parameter until activation time. For example:
+
+```js
+document.addEventListener("prerenderingchange", () => {
+  const via = (new URLSearchParams(location.search)).get("via");
+  analytics.send("via", via);
+});
+```
+
+### Interaction with redirects
+
+TODO
+
+### Interaction with storage partitioning
+
+This proposal takes place fully within the bounds of the [client side storage partitioning](https://github.com/privacycg/storage-partitioning/) work. It is only about calculating the cache key within a given partition. It does not affect the computation of the partition key itself.
+
+## Alternatives considered
+
+TODO
+
+## Future extensions
+
+### More complex no-vary rules
+
+As mentioned in the [non-goals section](#non-goals), it's possible to imagine a variety of complex matching rules for treating different query strings equivalently. The structured header format we're using allows some extensibility in this regard, if we decide such non-goals should become goals in the future. For example:
+
+- We could allow case-insensitivity of a given query parameter, using syntax like `No-Vary-Query: "color";value-case-insensitive=?1`.
+
+- We could allow restrictions on the value space, using syntax like `No-Vary-Query: "color";regexp="(?:blue|azure)"`.
+
+- We could allow treating multiple keys as the same key, using syntax like `No-Vary-Query: ("color", "colour")`.
+
+We think capabilities such as these serve less urgent [use cases](#use-cases) than what we have so far, and so don't plan on supporting them now. But it's good to know there's room for them in the future.
+
+### `No-Vary-Path`
+
+Many of the arguments for avoiding cache misses on differing query parameters, also apply to avoiding cache misses on differing paths. This is most prominent in the case of client-rendered single-page applications, where e.g. `/products/123` and `/products/456` might both have the same HTTP response, containing the skeleton product page, with their differing content being done via client-side rendering.
```

Comment:
Even though the web doesn't provide a total security boundary, it's some shared hosting providers (e.g., university departments) often separate different authors by path alone (more often than by query alone). That's not to say this couldn't be done eventually, but it merits a little additional thought on the risk/return tradeoff.

Reply:
At first I thought this was too weak of a consideration to include, but then I remembered service workers did indeed add a path restriction. So I've incorporated this. Thanks!